tests/pytest/oauth/test_views.py

```diff
@@ -99,76 +98,6 @@ def test_oauth_client_or_error_oauth_client(
     mocked_sentry_sdk_module.capture_exception.assert_not_called()
 
 
-@pytest.mark.django_db
-@pytest.mark.usefixtures("mocked_session_flow_uses_claims_verification", "mocked_oauth_client_or_error_redirect__error")
-def test_login_oauth_client_init_error(app_request, mocked_view_analytics_module):
-    result = login(app_request)
-
-    assert result.status_code == 302
-    assert result.url == reverse(routes.OAUTH_SYSTEM_ERROR)
-    mocked_view_analytics_module.error.assert_called_once()
-
-
-@pytest.mark.django_db
-def test_login_no_session_flow(app_request):
-    result = login(app_request)
-
-    assert result.status_code == 200
-    assert result.template_name == TEMPLATE_USER_ERROR
-
-
-@pytest.mark.django_db
-@pytest.mark.usefixtures("mocked_session_flow_uses_claims_verification")
-def test_login(app_request, mocked_oauth_client_or_error_redirect__client, mocked_view_analytics_module):
-    assert not session.logged_in(app_request)
-    mocked_oauth_client = mocked_oauth_client_or_error_redirect__client.return_value
-    # fake a permanent redirect response from the client
-    mocked_oauth_client.authorize_redirect.return_value.status_code = 301
-
-    login(app_request)
-
-    mocked_oauth_client.authorize_redirect.assert_called_with(app_request, "https://testserver/oauth/authorize")
-    mocked_view_analytics_module.started_sign_in.assert_called_once()
-    assert not session.logged_in(app_request)
```

Comment:
This one I think is difficult to keep.

The test would be good to keep because it covers that our session `logged_in` variable should be updated when the `login` view is called, but the problem is that in order to get through the `login` view successfully, we'd have to mock internal implementation details of the cdt_identity `login` view.

There are unit tests for the `OAuthHook` methods that assert `session.logged_in` was updated, and cdt_identity itself has unit tests to assert that the hooks were called in the right places, so overall we have enough coverage to justify not having this test.